### PR TITLE
Remove codalabworker package

### LINKED ISSUE #1346: Get rid of codalabworker package and just have one codalab package
This will simplify the currently convoluted dependencies between codalab and codalabworker.

### PR DESCRIPTION
**Pull request by candicegjing**
Fixed #1346 .

### CODE REVIEW

File: codalab/rest/workers.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def code():
     response.set_header("Content-Disposition", 'attachment; filename="code.tar.gz"')
     response.set_header("Content-Type", "application/gzip")
     codalab_cli = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
-    code_dir = os.path.join(codalab_cli, "worker", "codalabworker")
```
Comment (percyliang) [inline codalab/rest/workers.py:197]: This whole function is no longer used, so just get rid of it.

File: docker/dockerfiles/Dockerfile.worker [hunk 2]
```diff
@@ -6,22 +6,25 @@ ENV DEBIAN_FRONTEND noninteractive
...
+COPY setup.py setup.py
 
-# Install dependencies again
-RUN python3.6 -m pip install --user --upgrade pip; \
-    python3.6 -m pip install -e /opt/worker;
+RUN python3.6 -m pip install -e .
 
 # Allow non-root to read everything
-RUN chmod -R og=u-w /opt/worker
+RUN chmod -R og=u-w codalab/worker
```
Comment (percyliang) [inline docker/dockerfiles/Dockerfile.worker]: I'd do `.` instead of `codalab/worker`

File: pre-commit.sh [hunk 3]
```diff
@@ -2,13 +2,23 @@
 
 # Run this script before you commit.
 
+<<<<<<< HEAD
```
Comment (percyliang) [inline pre-commit.sh]: Need to resolve merge conflict

File: pre-commit.sh [hunk 4]
```diff
@@ -2,13 +2,23 @@
...
 if ! [ -e venv ]; then
   virtualenv -p python3.6 venv || exit 1
   venv/bin/pip install -r requirements-server.txt || exit 1
   venv/bin/pip install -r requirements.docs.txt || exit 1
   rm -rf worker/codalabworker.egg-info  # Need to clear because of different Python versions
   # Install for generating docs.
   venv/bin/pip install black==18.9b0 || exit 1
+=======
+if ! [ -e venv3.6 ]; then
+  virtualenv -p python2.7 venv3.6 || exit 1
```
Comment (percyliang) [inline pre-commit.sh]: no more 2.7

File: pre-commit.sh [hunk 5]
```diff
@@ -2,13 +2,23 @@
 
 # Run this script before you commit.
 
+<<<<<<< HEAD
 if ! [ -e venv ]; then
   virtualenv -p python3.6 venv || exit 1
   venv/bin/pip install -r requirements-server.txt || exit 1
   venv/bin/pip install -r requirements.docs.txt || exit 1
   rm -rf worker/codalabworker.egg-info  # Need to clear because of different Python versions
```
Comment (percyliang) [inline pre-commit.sh]: Don't need this line anymore

**Review by percyliang — CHANGES_REQUESTED** [review]
Mostly looks good.

File: codalab/worker/local_run/local_dependency_manager.py [hunk 6]
```diff
@@ -186,7 +186,7 @@ def _cleanup(self):
             with self._global_lock:
                 self._acquire_all_locks()
                 bytes_used = sum(dep.size_bytes for dep in self._dependencies.values())
-                serialized_length = len(codalabworker.pyjson.dumps(self._dependencies))
```
Comment (epicfaace) [inline codalab/worker/local_run/local_dependency_manager.py:189]: Should this be codalab.worker.pyjson?

File: scripts/upload-to-pypi.sh [hunk 7]
```diff
@@ -9,13 +9,8 @@ usage()
...
 VER=$1
 
-echo "==> Packaging codalabworker"
-cd ./worker
-python3.6 setup.py sdist
-echo "==> Uploading codalabworker"
-twine upload dist/codalabworker-$VER.tar.gz
 echo "==> Packaging codalab"
-cd ..
+cd ./
```
Comment (epicfaace) [inline scripts/upload-to-pypi.sh]: Could we just remove this line?

File: worker/setup.py [hunk 8]
```diff
@@ -1,42 +0,0 @@
...
-                if (
-                    line[0:2] != '-r'
-                    and line.find('git') == -1
-                    and line.find('codalabworker') == -1
-                ):
-                    requirements.append(line)
-    return requirements
-
-
-setup(
```
Comment (epicfaace) [inline worker/setup.py:21]: Should we publish some kind of notice to pypi that codalabworker is deprecated and should be replaced by codalab?

**Comment by percyliang** [comment]
Another thing: can you merge `codalab/worker/formatting.py` into `codalab/lib/formatting.py` (same with `file_util.py` and any other utility)?

File: codalab/worker/local_run/local_dependency_manager.py (hunk 6 above)
Comment (candicegjing) [reply codalab/worker/local_run/local_dependency_manager.py:189]: Hmm, this looks interesting... I double-checked my local branch and also the Github version which was committed 21 hours ago: https://github.com/codalab/codalab-worksheets/blob/remove-codalabworker-package/codalab/worker/local_run/local_dependency_manager.py#L189. Both of them have `codalabworker` replaced with `codalab.worker`. Not sure what was happening here...

File: codalab/rest/workers.py (hunk 1 above)
Comment (candicegjing) [reply codalab/rest/workers.py:197]: Hmm, I am actually not sure what was happening here. I double-checked my local branch and also the version that I committed to Github 21 hours ago: https://github.com/codalab/codalab-worksheets/blob/remove-codalabworker-package/codalab/rest/workers.py. None of them have this line. It seems that @epicfaace also observed a similar issue but in a different file. But if this `def code()` is not used, I'll remove it.

File: scripts/upload-to-pypi.sh (hunk 7 above)
Comment (candicegjing) [reply scripts/upload-to-pypi.sh]: Yes, I'll remove it.

File: pre-commit.sh (hunk 3 above)
Comment (candicegjing) [reply pre-commit.sh]: I'll resolve conflicts in this file.

File: codalab/rest/workers.py (hunk 1 above)
Comment (percyliang) [reply codalab/rest/workers.py:197]: Yeah, remove the whole function.

**Comment by candicegjing** [comment]
I am closing this PR as it seems to be a `partial-pull-merging`

**Comment by epicfaace** [comment]
What's a `partial-pull-merging`?

**Pull request closed without merging**